src/read.jl

```diff
@@ -475,6 +500,31 @@ function reformat_demand_flexibility_input(
             # store to new df object
             demand_flexibility_updated[field] = eq_bus_df
 
+            # if flex amount specified in DOE data, re-format using bus EIA ID
+        elseif demand_flexibility.source == "doe" &&
+                field in ["flex_amt_up", "flex_amt_dn"]
+            
+            # all load buses with valid EIA ID 
+            flexible_bus_idx = intersect(sets.load_bus_idx, findall(case.bus_eiaid .> 0))
+            flexible_bus_num = length(flexible_bus_idx)
+            
+            # bus flexibility Percentage matrix
+            doe_bus_flexibility = zeros(size(case.demand, 1), flexible_bus_num)
+            doe_eiaid = [parse(Int, x) for x in names(fh)[2:end]]
+            for i in 1:flexible_bus_num
+                if case.bus_eiaid[flexible_bus_idx[i]] in doe_eiaid
+                    doe_bus_flexibility[:, i] = fh[Symbol(case.bus_eiaid[flexible_bus_idx[i]])]
+                end 
```

Comment:
Do we want to warn the user somehow if a bus has an EIA ID that doesn't have a corresponding entry in the flexibility data?

Reply:
Yes, a check will be added later. In the Texas subsystem I am using now there are no such buses so it does not prompt an error, but this will be a problem for the USA network. There are like 11 load buses in the USA system that are located in Canada and Mexico that do not have an EIAID associated with them. They are marked as 0 for now and in the future we might want to assign them to their nearest neighbor.